### PR TITLE
Transition to default int4 compression configs from optimum-intel

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
**Changes**

- For `4BIT_DEFAULT`, the compression config is now always imported from optimum-intel
- Remove all default configs from openvino.genai, rely only on default configs in optimum-intel
- Adopt dataset preparation logic from optimum-intel

Breaking changes:
- Old compression dataset specification is deprecated. For example `"wikitext,wikitext-2-v1,train[:1000],text"` will not work anymore. A string `"wikitext2"` should be used instead.
- Compression with data-aware methods will produce different results, potentially worse.
- Default int4 configs will no longer be matched by folder names. To match the default config, one of these options must hold:
  - model is exported with `python convert.py -m <model_id>`
  - model is exported with `python convert.py -m <model_path>`, where `model_path` contains a `config.json` file with a `_name_or_path` field containing a model id
  - model is exported with `python convert.py -m <model_path>`, where `model_path` contains a `config.json` file with a `_name_or_path` field containing another `model_path_2`, such that the condition from the bullet point above holds for it

**Related tickets**
147470

### CODE REVIEW

File: llm_bench/python/utils/conversion_utils/helpers.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def get_data_aware_args(ov_model, tokenizer, config, compression_args, args):
         dataset_args = compression_args['dataset']
         dataset_params = dataset_args['name']
         if 'sensitivity_metric' in dataset_args:
-            res['mode'] = dataset_args['sensitivity_metric']
+            res['sensitivity_metric'] = dataset_args['sensitivity_metric']
```
Comment (nikita-savelyevv) [inline llm_bench/python/utils/conversion_utils/helpers.py:160]: @andreyanufr it shouldn't be `mode`, right?

File: llm_bench/python/utils/conversion_utils/helpers.py [hunk 2]
```diff
@@ -189,17 +189,37 @@ def compress_ov_model_weights_helper(ov_model, tok, config, out_path, compress_w
...
+        sensitivity_metric = compression_args.pop("sensitivity_metric", None)
+        num_samples = compression_args.pop("num_samples", None)
+        if num_samples:
+            compression_args["subset_size"] = num_samples
+        dataset = compression_args.pop("dataset", None)
+        if dataset:
+            if dataset == "wikitext2":
+                dataset_name = "wikitext,wikitext-2-v1,train[:1000],text"
+            elif dataset == "c4" or dataset == "c4-new":
+                dataset_name = "allenai/c4,en,train,text"
```
Comment (nikita-savelyevv) [inline llm_bench/python/utils/conversion_utils/helpers.py:209]: optimum-intel currently supports only these 3 string datasets

File: llm_bench/python/utils/conversion_utils/helpers.py [hunk 3]
```diff
@@ -96,8 +96,8 @@ def save_tokenizer(tokenizer, out_dir):
 
 def transform_fn(item, item_name, input_shapes, tokenizer, config, max_tokens=127):
     tokenized_text = tokenizer(item[item_name], return_tensors="np")
-    input_ids = tokenized_text["input_ids"][:max_tokens]
-    attention_mask = tokenized_text["attention_mask"][:max_tokens]
+    input_ids = tokenized_text["input_ids"][:, :max_tokens]
+    attention_mask = tokenized_text["attention_mask"][:, :max_tokens]
```
Comment (nikita-savelyevv) [inline llm_bench/python/utils/conversion_utils/helpers.py:100]: Adopt changes from #672

File: llm_bench/python/utils/nncf_utils.py [hunk 4]
```diff
@@ -35,45 +35,3 @@
 
 def get_compressed_path(output_dir: str, base_precision, option: str):
     return Path(output_dir) / "pytorch/dldt/compressed_weights" / f"OV_{base_precision}-{option}"
-
-
-INT4_MODEL_CONFIGURATION = {
```
Comment (nikita-savelyevv) [inline llm_bench/python/utils/nncf_utils.py:40]: Yet need to resolve some config misallignments:

- For `databricks/dolly-v2-3b` in optimum `ratio` is 1 while in genai is 0.8.

- Some configs from genai are missing in optimum:
```
red-pajama-incite-7b-instruct
llama-2-7b
llama-2-7b-chat
llama-2-13b-chat
stablelm-epoch-3b-preview
stable-zephyr-3b-dpo
rocket-3b
qwen-7b-chat
open-llama-3b
orca-mini-3b
mixtral-8x7b-v0.1
baichuan2-7b-chat
mistral-7b-v0.1
llama-7b
red-pajama-incite-chat-3b-v1
longchat-b7
tiny-llama-1.1b-chat
```

- Some configs in optimum are missing in genai:
```
togethercomputer/RedPajama-INCITE-7B-Instruct
meta-llama/Llama-2-7b
meta-llama/Llama-2-7b-chat
meta-llama/Llama-2-13b-chat
stabilityai/stablelm-zephyr-3b
pansophic/rocket-3B
Qwen/Qwen-7B-Chat
openlm-research/open_llama_3b
openlm-research/open_llama_3b_v2
psmathur/orca_mini_3b
mistralai/Mixtral-8x7B-v0.1
togethercomputer/RedPajama-INCITE-Chat-3B-v1
mistralai/Mistral-7B-v0.1
baichuan-inc/Baichuan2-7B-Chat
openai-community/gpt2
lmsys/longchat-7b-16k
TinyLlama/TinyLlama-1.1B-Chat-v1.0
stabilityai/stablelm-tuned-alpha-7b
```

File: llm_bench/python/utils/conversion_utils/helpers.py (hunk 1 above)
Comment (andreyanufr) [reply llm_bench/python/utils/conversion_utils/helpers.py:160]: res['sensitivity_metric'] = dataset_args['sensitivity_metric'] is right option

File: llm_bench/python/utils/nncf_utils.py (hunk 4 above)
Comment (nikita-savelyevv) [reply llm_bench/python/utils/nncf_utils.py:40]: Resolved

**Review by eaidova — APPROVED** [review]

**Review by skuros — APPROVED** [review]

**Pull request merged**